src/inmanta/agent/cache.py

```diff
@@ -40,25 +40,31 @@ def __init__(
         timeout: float,
         value: Any,
         call_on_delete: Optional[Callable[[Any], None]],
-        lingering: bool = True,
+        evict_after_last_access: bool = True,
+        evict_after_creation: bool = False,
     ) -> None:
         """
         :param key: The full key identifying this item in the cache.
-        :param timeout: Hard timeout for non-lingering cache item.
+        :param timeout: Hard timeout used in combination with evict_after_creation=True.
         :param value: The value being cached associated to the key.
         :param call_on_delete: Optional finalizer to call when the cache item is deleted. This is
             a callable expecting the cached value as an argument.
-        :param lingering: When True, this cache item will linger in the cache for 60s after its last use.
-            When False, this cache item will be evicted from the cache <timeout> seconds after
+        :param evict_after_last_access: When True, this cache item will stay in the cache for 60s after its last use.
+        :param evict_after_creation: When True, this cache item will be evicted from the cache <timeout> seconds after
             entering the cache.
         """
         self.key = key
         self.value = value
         self.call_on_delete = call_on_delete
-        self.lingering = lingering
+        self.evict_after_last_access = evict_after_last_access
```

Comment:
I think I would name it like this
```suggestion
        self.referesh_after_access = evict_after_last_access
```

Reply:
I like this suggestion :+1: